src/lsp/DefinitionProvider.ts

```diff
@@ -0,0 +1,39 @@
+import * as vscode from "vscode";
+import { createLogger } from "../utils";
+
+const log = createLogger("lsp.definitions");
+
+export class DefinitionProvider implements vscode.DefinitionProvider {
+
+	public data: Map<string, string> = new Map();
+
+	constructor(private client) {
+		vscode.languages.registerDefinitionProvider({ scheme: "file", language: "gdscript" }, this);
+	}
+
+	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Definition> {
+		const key = `${document.uri},${position.line},${position.character}`;
```

Comment:
This is where the intermittent issue come from when go to definition doesn't open the docs.

Taking inspiration from my original PR, you could instead get the symbol via a mock `textDocument/hover` message. I'd personally extract the symbol parsing logic from `handle_hover_response` into its own method and then you can return the symbol from an async `get_symbol_at_position` method called from `DefinitionProvider::provideDefinition`

```typescript
// GDScriptLanguageClient.ts
public async get_symbol_at_position(uri: vscode.Uri, position: vscode.Position): string {
    const params = {
	  textDocument: { uri: uri.toString() },
	  position: { line: position.line, character: position.character },
    };
    const response = await this.sendRequest("textDocument/hover", params);
    return this.parse_hover_response(response)
}
```

```typescript
// DefinitionProvider.ts
export class DefinitionProvider implements vscode.DefinitionProvider {
    
    constructor(private client: GDScriptLanguageClient) {}
    
    async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Definition> {
        const symbol = await this.client.get_symbol_at_position(document.uri, position);
        if (!symbol) {
            return null;
        }
        
        const parts = symbol.split(".");
        const uri = vscode.Uri.from({
            scheme: "gddoc",
            path: parts[0] + ".gddoc",
            fragment: parts[1],
        });
        
        return new vscode.Location(uri, new vscode.Position(0, 0));
    }
}
```

```typescript
// ClientConnectionManager.ts
public client: GDScriptLanguageClient = null;
public definitionProvider: DefinitionProvider = null;

constructor(private context: vscode.ExtensionContext) {
    ...
    this.client = new GDScriptLanguageClient(context);
    this.definitionProvider = new DefinitionProvider(this.client);
    vscode.languages.registerDefinitionProvider({ scheme: "file", language: "gdscript" }, this.definitionProvider);
    ...
}
```

It seems to work pretty flawlessly from the little testing I've done so far.